<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>969</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Cpp0x tags editor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="4" column="1">
     <widget class="QGroupBox" name="contextGroup">
      <property name="title">
       <string>Short conspect</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QTableWidget" name="contextTableWidget">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="alternatingRowColors">
          <bool>false</bool>
         </property>
         <property name="cornerButtonEnabled">
          <bool>false</bool>
         </property>
         <column>
          <property name="text">
           <string>line</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>tag</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>text</string>
          </property>
         </column>
        </widget>
       </item>
       <item>
        <widget class="GoToLineWidget" name="goToLineGroupBox">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>70</height>
          </size>
         </property>
         <property name="title">
          <string>Go to line</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="6" column="0" colspan="2">
     <widget class="FindDialog" name="findWidget" native="true"/>
    </item>
    <item row="2" column="1">
     <widget class="ErrorList" name="errorsInText" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="MinimumExpanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="StcTagsButtons" name="buttonsEmittingStc">
      <property name="frameShape">
       <enum>QFrame::Shape::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
     </widget>
    </item>
    <item row="0" column="0" rowspan="6">
     <widget class="CodeEditor" name="plainTextEdit">
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QPlainText {
	background-color: rgb(255, 255, 255);
	color: rgb(0, 0, 0);
    font-family: monospace;
}

</string>
      </property>
     </widget>
    </item>
    <item row="5" column="1">
     <widget class="QLabel" name="label">
      <property name="text">
       <string extracomment="link do kursu">&lt;a href=&quot;https://cpp0x.pl/kursy/Kurs-STC/169&quot;&gt;Kurs STC&lt;/a&gt;</string>
      </property>
      <property name="textFormat">
       <enum>Qt::TextFormat::RichText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
      <property name="openExternalLinks">
       <bool>true</bool>
      </property>
      <property name="textInteractionFlags">
       <set>Qt::TextInteractionFlag::TextBrowserInteraction</set>
      </property>
     </widget>
    </item>
    <item row="3" column="1">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_file"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_as"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionFind"/>
    <addaction name="actionFind_replace"/>
   </widget>
   <widget class="QMenu" name="menuCheck">
    <property name="title">
     <string>Check</string>
    </property>
    <addaction name="actioncheck_if_tags_are_closed"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuCheck"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionLoad_file">
   <property name="text">
    <string>Load file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="text">
    <string>Save file as</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionFind">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Find</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="actionFind_replace">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Find and replace</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actioncheck_if_tags_are_closed">
   <property name="text">
    <string>check if tags are closed</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CodeEditor</class>
   <extends>QPlainTextEdit</extends>
   <header location="global">codeeditor.h</header>
  </customwidget>
  <customwidget>
   <class>FindDialog</class>
   <extends>QWidget</extends>
   <header location="global">ui/finddialog.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ErrorList</class>
   <extends>QWidget</extends>
   <header location="global">ui/errorlist.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>GoToLineWidget</class>
   <extends>QGroupBox</extends>
   <header>ui/gotolinewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>StcTagsButtons</class>
   <extends>QFrame</extends>
   <header>ui/stctagsbuttons.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>plainTextEdit</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>actionSave_as</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onSaveAsPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>389</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoad_file</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onOpenPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>443</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFind</sender>
   <signal>toggled(bool)</signal>
   <receiver>findWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>866</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actioncheck_if_tags_are_closed</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onCheckTagsPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>451</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onSavePressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>451</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onSaveAsPressed()</slot>
  <slot>onOpenPressed()</slot>
  <slot>onFindCheckChanged(bool)</slot>
  <slot>onCheckTagsPressed()</slot>
  <slot>onSavePressed()</slot>
 </slots>
</ui>
